doc/rst/source/develop.rst

```diff
@@ -0,0 +1,101 @@
+   the new module's name and purpose. Then there is the **THIS_MODULE_OPTIONS* parameter which
+   is just a listing of all the common GMT options this module can use.  The next item is
+   **THIS_MODULE_NEEDS** and it tells GMT if this module requires a region (e.g., via **-R**)
+   and a projection (via **-J**).  Modules that must set a region to operate will either have
+   the codes **R** (must take **-R**), **d** (can determine the region from given data sets),
+   or **g** (can determine the region from a given grid).  There is also the mode **r** which
+   means 'not normally required' but may be required depending on module options.  As an example,
+   consider :doc:`pslegend` which does not need **-R** if **-Dx** is used but *does* need **-R** if
+   other selections are made. For the same reason there is the **j** code for projections when
+   it depends on other options whether **-J** is needed or not.  Most plot modules NEEDS will
+   include **J**.  Finally, there is the very cryptic **THIS_MODULE_KEYS** setting. At this point,
+   please read all the comments in the code for *GMT_Encode_Options* in gmt_api.c.  The **KEYS**
+   are not used by the command line modules and only enters for some external interfaces, such
+   as MATLAB and Julia.  You should have a discussion with those maintainers about what those
+   **KEYS** should be.
+
+#. Do some global replacements of strings: Replace the uppercase name of the template module
+   with the new module uppercase name, and do the same for the lowercase name. This will ensure that
+   your Ctrl structures and local static functions are named correctly (you may still need to
+   delete some of them and add others).  All functions local to this module shall have names
+   starting with the module name and an underscores (e.g., grdinfo_report_tiles*) and they shall
+   be stated as GMT_LOCAL instead of static (GMT_LOCAL is normally defined to mean static but
+   in certain debugging cases we may wish to change that). The exceptions to this rule are the
+   functions* New_Ctrl*, *Free_Ctrl*, *usage*, and *parse*; they are all static and have the
+   same names in all modules.
+
+#. After listing all local variables inside the main GMT_module function, all modules start with
+   a section labeled *Standard module initialization and parsing*.  It is extremely unlikely
+   you will need to make any changes in this section.  Ask for help if you think you do.
+
+#. For style, argument checking, usage layout, use of built-in constants for the synopsis line,
+   just follow the module template and draw inspiration from other modules.
+
+#. Make sure you create a separate git branch and add your new code to that branch. New modules
+   should never go into master and will have to undergo much scrutiny before the branch may
+   be merged into master.
+
+#. Try to compile GMT with your new module.  We automatically detect all the modules in the *src*
```

Comment:
Have to update **GMT_PROGS_SRCS** in src/CMakeLists.txt.

Reply:
OK, added